# gene-expression

[COMPLETED] How do two dominant grasses differ in their gene expression responses to drought?

Grasslands of the Central US are expected to experience severe droughts and other climate extremes in the future, yet we know little about how these grasses will respond in terms of gene expression. 

We compared gene expression in Andropogon gerardii and Sorghastrum nutans, two closely related co-dominant C4 grasses responsible for the majority of ecosystem function, using RNA-seq. We compared Trinity assemblies within each species to determine annotated functions of transcripts responding to drought. Subsequently, we compared homologous annotated gene-groups across the two species using cross-species meta-level analysis and functional clustering based on key terms. 

The majority of variation was found between species, as opposed to between drought and watered treatments. However, there is evidence for differential responses; Andropogon allocated gene expression differently compared to Sorghastrum, suggesting Andropogon focuses on stress alleviation (such as oxygen radical scavenging) rather than prevention. In contrast, Sorghastrum may employ a drought avoidance strategy by modulating osmotic response, especially with hormonal regulation. We found Sorghastrum tended to be more sensitive within 10 key gene-groups related to stress, abscisic acid, and trichomes, suggesting gene expression may mechanistically parallel sensitivity at the physiological level. 

Our findings corroborate phenotypic and physiological differences in the field, and may help explain the phenotypic mechanisms of these two species in the tallgrass prairie community under future drought scenarios.  

**Data available at: https://datadryad.org//resource/doi:10.5061/dryad.5bk4c**
